semantic/graph.go

```diff
@@ -934,7 +934,7 @@ func (l *BooleanLiteral) Copy() Node {
 	return nl
 }
 func (e *BooleanLiteral) TypeOf() MonoType {
-	return e.typ
+	return BasicBool
```

Comment:
I would prefer if `TypeOf` returned the inferred type. I'd like to know if we ever infer anything other than `bool`.

Reply:
In the future we can remove this. The problem is resolving functions currently replaces values with semantic nodes and so there's no type inference done. We can hopefully remove this in the future because I think you're right.